### PR TITLE
add warning logs for storage error

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

Add warning logs for storage operation error. Also added test cases to cover exception branches.

## Motivation and Context

To get better understanding what went wrong.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [x] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

**Comment by danielnorberg** [comment]
This could be a middleware that logs internal server errors instead of manually adding this code in every method handler.

**Comment by honnix** [comment]
Good point. Now I remember ResponseException, maybe we should align all the different usages of response and exception in API. It's a bit messy now.

**Comment by danielnorberg** [comment]
Yeah, it might be good to have less manual error handling in the handler methods.

**Comment by honnix** [comment]
@danielnorberg Pushed minimal change. PTAL

File: styx-service-common/src/main/java/com/spotify/styx/api/Middlewares.java [hunk 1]
```diff
@@ -150,8 +151,10 @@ private Middlewares() {
         return completedFuture(e.<T>getResponse()
             .withHeader(X_REQUEST_ID, requestId));
       } catch (Throwable t) {
+        var internalServerErrorReason = internalServerErrorReason(requestId, t);
```
Comment (honnix) [inline styx-service-common/src/main/java/com/spotify/styx/api/Middlewares.java]: Annoying we don't follow the same pattern in our APIs.

**Comment by danielnorberg** [comment]
LGTM 👍

**Pull request merged**